lib/screens/home_screen.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:qrdemo/providers/qr_provider.dart';
+import 'package:qrdemo/providers/routing_provider.dart';
+import 'package:qrdemo/providers/timer_provider.dart';
+import 'package:simple_speed_dial/simple_speed_dial.dart';
+
+class HomeScreen extends StatelessWidget {
+  const HomeScreen({Key? key}) : super(key: key);
+
+  Future<void> reset(BuildContext context) async {
+    Provider.of<TimerProvider>(context, listen: false).setExpire(false);
+    Provider.of<TimerProvider>(context, listen: false).reset();
+    Provider.of<QRProvider>(context, listen: false).generateCode();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<void>(
+      future: reset(context),
```

Comment:
The build method here only runs when the user is navigated to the homescreen. I did this because the timer kept overlapping and I had to force a reset on the timer every time the user goes to homescreen. The timer was still active and when the user goes to generate a QR, the timer will go two increments faster. Doing this approach worked best for me to handle that problem. Do you suggest a different way?

Reply:
FutureBuilder only builds the UI once the future has been called and depending on the connection state of the snapshot, it will render the UI accordingly.